### PR TITLE
Added support for multiple secrets

### PR DESCRIPTION
**Pull request by KavishShah09**

### CODE REVIEW

File: gateway/modules/auth/auth.go [hunk 1]
```diff
@@ -205,3 +207,15 @@ func (m *Module) SetMakeHTTPRequest(function utils.MakeHTTPRequest) {
...
+func (m *Module) getGreatestSecretKey() int {
+	keys := make([]int, 0)
+	for key := range m.secrets {
+		keys = append(keys, key)
+	}
+	sort.Ints(keys)
+	if len(keys) == 0 {
+		return 0
+	}
+	return keys[len(keys)-1]
```
Comment (YourTechBud) [inline gateway/modules/auth/auth.go:220]: Find a better way to do this

File: gateway/modules/auth/auth_test.go [hunk 2]
```diff
@@ -60,22 +62,22 @@ func TestParseToken(t *testing.T) {
...
-		{name: "Test should fail for an invalid token", IsErrExpected: true, secretKey: "mysecretkey", token: "1234.abcd"},
-		{name: "Test should fail for invalid signature-illegal base64 data at input", IsErrExpected: true, secretKey: "mySecretkey", token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbjEiOiJ0b2tlbjF2YWx1ZSIsInRva2VuMiI6InRva2VuMnZhbHVlIn0.h3jo37fYvnf55A63N"},
-		{name: "Test should fail as invalid secret key-invalid signature", IsErrExpected: true, secretKey: "mysecretkey", token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbjEiOiJ0b2tlbjF2YWx1ZSIsInRva2VuMiI6InRva2VuMnZhbHVlIn0.h3jo37fYvnf55A63N-uCyLj9tueFwlGxEGCsf7gCjDc"},
+		{name: "Test should successfully parse a token", IsErrExpected: false, secretKeys: map[int]string{0: "mySecretkey"}, token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbjEiOiJ0b2tlbjF2YWx1ZSIsInRva2VuMiI6InRva2VuMnZhbHVlIn0.h3jo37fYvnf55A63N-uCyLj9tueFwlGxEGCsf7gCjDc", wantThis: TokenClaims{"token1": "token1value", "token2": "token2value"}},
```
Comment (YourTechBud) [inline gateway/modules/auth/auth_test.go:70]: Add one more test case similar to this with multiple secrets. The current secret should be defined somewhere in the middle.

File: gateway/config/config.go [hunk 3]
```diff
@@ -9,13 +9,13 @@ type Config struct {
...
 // Project holds the project level configuration
 type Project struct {
-	Secret         string   `json:"secret,omitempty" yaml:"secret,omitempty"`
-	AESkey         string   `json:"aesKey,omitempty" yaml:"aesKey,omitempty"`
-	ID             string   `json:"id,omitempty" yaml:"id,omitempty"`
-	Name           string   `json:"name,omitempty" yaml:"name,omitempty"`
-	DockerRegistry string   `json:"dockerRegistry,omitempty" yaml:"dockerRegistry,omitempty"`
-	Modules        *Modules `json:"modules,omitempty" yaml:"modules,omitempty"`
-	ContextTime    int      `json:"contextTime,omitempty" yaml:"contextTime,omitempty"` // contextTime sets the timeout of query
+	Secrets        map[int]string `json:"secret,omitempty" yaml:"secret,omitempty"`
```
Comment (YourTechBud) [inline gateway/config/config.go:12]: Change the annotations

File: gateway/config/load.go [hunk 4]
```diff
@@ -53,3 +55,12 @@ func LoadConfigFromFile(path string) (*Config, error) {
...
 	return conf, nil
 }
+
+func getGreatestSecretKey(secrets map[int]string) int {
+	keys := make([]int, 0)
+	for key := range secrets {
+		keys = append(keys, key)
+	}
+	sort.Ints(keys)
+	return keys[len(keys)-1]
```
[the remark this answers is not in the document]
Comment (KavishShah09) [reply gateway/config/load.go:65]: Only used at one place

[the remark this answers is not in the document]
Comment (KavishShah09) [reply gateway/config/load.go:65]: Infact not even used here anymore

**Comment by YourTechBud** [comment]
Resolve the conflicts in this branch @KavishShah09

File: gateway/config/load.go [hunk 5]
```diff
@@ -2,29 +2,32 @@ package config
...
+				if present {
+					p.Secrets[key] = tempEnvVar
+				}
 			}
-		}
-		for _, value := range p.Modules.Crud {
-			if strings.HasPrefix(value.Conn, "$") {
-				tempStringC := strings.TrimPrefix(value.Conn, "$")
-				tempEnvVarC, presentC := os.LookupEnv(tempStringC)
+			for _, value := range p.Modules.Crud {
```
Comment (YourTechBud) [inline gateway/config/load.go:23]: Take this out of the secrets loop

File: gateway/modules/auth/auth.go [hunk 6]
```diff
@@ -205,3 +207,15 @@ func (m *Module) SetMakeHTTPRequest(function utils.MakeHTTPRequest) {
 
 	m.makeHTTPRequest = function
 }
+
+func (m *Module) getGreatestSecretKey() string {
+	greatesKey := 0
+	for key := range m.secrets {
+		intKey, _ := strconv.Atoi(key)
```
Comment (YourTechBud) [inline gateway/modules/auth/auth.go:214]: Check for error. Ignore lets with error

File: gateway/modules/auth/auth.go [hunk 7]
```diff
@@ -205,3 +207,15 @@ func (m *Module) SetMakeHTTPRequest(function utils.MakeHTTPRequest) {
 
 	m.makeHTTPRequest = function
 }
+
+func (m *Module) getGreatestSecretKey() string {
```
Comment (YourTechBud) [inline gateway/modules/auth/auth.go:211]: Write test for this method

**Pull request merged**